haystack/document_store/base.py

```diff
@@ -324,3 +325,19 @@ def _handle_duplicate_documents(self, documents: List[Document], duplicate_docum
             documents = list(filter(lambda doc: doc.id not in ids_exist_in_db, documents))
 
         return documents
+
+    def _get_duplicate_labels(self, labels: list, index: str = None) -> List[Label]:
+        """
+        Return all duplicate labels
+        :param labels: List of Label objects
+        :param index: add an optional index attribute to labels. It can be later used for filtering.
+        :return: List of labels
+        """
+        index = index or self.label_index
+        ids = [label.id for label in labels]
+        duplicate_ids = [label_id for label_id, count in collections.Counter(ids).items() if count > 1]
```

Comment:
This is quite a tricky line to follow because of how long it is. Could you separate this out into a few lines so that it is easier to debug in future?

Reply:
Done.